### DISCUSSION TITLE
How to specify @types dependencies?

### DISCUSSION THREAD

**Discussion by benknab** [DefinitelyTyped/DefinitelyTyped#54335] [_Misc]
I would like to add `@types` dependencies for a package and the README says I don't have to include these in a `package.json`, where else can I specify these type dependencies?
> A `package.json` may be included to specify dependencies that are not other `@types` packages.

**Comment by eritbh** [comment]
That section of the readme looks like it's talking about developing the `@types` packages themselves - the `@types` packages can have their own dependencies if the library definition relies on types from another package.

For installing a `@types` package for your own use, you generally install them to `devDependencies` in your project, e.g. via `npm install --save-dev @types/package`.

**Comment by benknab** [comment]
I'm also asking about developing `@types` packages, I wanted to create a package that uses types from other `@types` packages but I couldn't figure out how, sorry for not being clearer.

**Reply by eritbh** [reply]
I *think* what that's saying is that if you need a dependency that has a `@types` package, you can just import the package and the dependency stuff will be resolved for you. You only need to worry about putting dependencies in a `package.json` if you need to import types from a package that has its types built-in.

**Reply by benknab** [reply]
Thanks, I will try it out!
